src/printer.js

```diff
@@ -548,29 +548,46 @@ function genericPrintNoParens(path, options, print) {
         "body"
       );
 
-      // If there are no contents, return a simple block
-      if (!getFirstString(naked)) {
+      const hasDirectives = n.directives && n.directives.length > 0;
+      const hasContent = getFirstString(naked);
+      const hasInnerComments = n.innerComments && n.innerComments.length > 0;
+
+      if (!hasInnerComments && !hasDirectives && !hasContent) {
         return "{}";
       }
 
       parts.push("{");
 
-      // Babel 6
-      if (n.directives) {
+      // Babylon 6
+      if (hasInnerComments) {
+        path.each(function(innerComment) {
+          parts.push(
+            indent(
+              options.tabWidth,
+              concat([ hardline, print(innerComment) ])
+            )
+          );
+        }, "innerComments");
```

Comment:
I'd like to wait to merge this PR until Monday when I have more time to think about comments. Right now comments are always interleaved separated in the `printComments` function in `comments.js`. I think to fix various edge cases we are going to have to manually start printing comments like this in a few places, but I'd like to thorougly think it through before touching much of the comment code.